core/arch/x86/mangle.c

```diff
@@ -2403,7 +2405,17 @@ mangle_single_step(dcontext_t *dcontext, instrlist_t *ilist, instr_t *instr)
      * FIXME i#2144 : to be absolutely transparent, we should translate the
      * exception address as if we did not insert this nop.
```

Comment:
> Using the jump target instead of the jump is unique to STATUS_SINGLE_STEP, right? 

Yes to the best of my knowledge.

The tests I added will not fail because my fault handler does not check the exception address (just the exception code)

I will try to put this check before `check_internal_exception`